Integrate donation circle with join us include

jenny-alexander opened the pull request:
Fixes #4544 

### What changes did you make and why did you make them ?

  - In _includes/join-donate-card.html replace the corresponding <img> tag with
```
        <div class="join-us-donation-circles">
          {%  include /donation/donation-circles.html %}
        </div>
```
  - In donation-circles.html, I added id="override-p-padding" to each p tag within the <div class="circle"> elements. This was done because there was padding-bottom being added from _cards.scss and I needed a way to override this padding so the text within the donation circles was padded according to the specifications.
`  <p id="override-p-padding" class="inside-circle-text">Hosting</p>`
  - In _sass/components/_donate-components.scss, I added a new id selector called override-p-padding that sets the padding to 0.2em 0em.
```
  #override-p-padding {
    padding: 0.2em 0em;
   }
```
_Although the initial request was to add padding of 0.4em, I added 0.2em since it was was requested by @Adastros and agreed upon by @one2code and @roslynwythe._

  - In _sass/components/_donate-components.scss, I removed `padding: 0.4em 0em` from class selector inside-circle-text since the padding is done within #override-p-padding now.

- Removed duplicate padding (`padding: 0`) within circle-full-container class in _donate-components.scss that was causing linter error.
   
### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
- I tested both Join Us and Donate pages since the donation circles are shared across both of those components. 


<details>
<summary>Visuals before changes are applied</summary>

Join Us - Desktop Breakpoint
![Desktop](https://github.com/hackforla/website/assets/80556079/d1076496-177c-4f9c-8b71-a893ca377441)

Join Us - Mobile Breakpoint
![Mobile](https://github.com/hackforla/website/assets/80556079/de718a44-eeba-477b-8045-de17c7de2aa7)

</details>

<details>
<summary>Visuals after changes are applied</summary>
Join Us - Desktop Breakpoint

![Screenshot 2023-05-30 at 8 00 46 PM](https://github.com/hackforla/website/assets/80556079/ed568a66-11ce-4c62-92bc-1cef84e56458)

Join Us - Mobile Breakpoint
![Mobile](https://github.com/hackforla/website/assets/80556079/3bc979ee-5f9c-49eb-9bba-4b563343b067)

</details>

- Additionally, see the two documents for testing results.
[Join Us testing.pdf](https://github.com/hackforla/website/files/11607623/Join.Us.testing.pdf)
[Donate testing.pdf](https://github.com/hackforla/website/files/11560081/Donate.testing.pdf)

jenny-alexander committed:
Integrate donation circles into join us as include

jenny-alexander committed:
Add space between selectors to keep formatting consistent

jenny-alexander commented:
The Lint SCSS job failed because it found duplicates for padding in .circle-full-container (in _donate-components.scss).

.circle-full-container {
    display: flex;
    flex-direction: column;
    **padding: 1em 2em;**
    justify-content: center;
    align-items: center;
    **padding: 0;**
    margin: 0;
}

This css portion wasn't introduced by me. Should I be the person to remove the padding at line 15 since it is being overwritten at line 18?

one2code commented:
**ETA**: 5/27/23

I want to delve in further and explore other solutions. I'm also interested in hearing from a team lead on their recommended solution to resolve the issue. If they are ok with the #id override, and removing the padding 0; that was added while addressing issue #4573, then I would potentially request this code change. I've tested it in Docker, and it works. It will save you from individually adding the id class to each p tag.

This can be added to the _donate-components.scss file
```
@for $i from 1 through length($inside-circle-items) {
    .inside-circle-text:nth-child(#{$i}) {
      content: nth($inside-circle-items, $i);
      
    }
```

Adastros commented:
Availability (PST): 11:45 PM 24MAY2023- 12:00 AM 25MAY2023
Review ETA (PST): 12:00 AM 25MAY2023

Adastros reviewed (changes requested):
Hi @jenny-alexander, the liquid tag was implemented correctly and works at all breakpoints for Firefox and Chrome. 

Below are my comments:

- [x] In regard to the lint error, you can remove `padding: 0` from class `.circle-full-container`. This will center the 3 circles under the text paragraph. I don't think this is outside the scope of the issue. 
- [x] In regards to the `.override-p-padding` class, I think it would be best to try implementing @one2code's solution in the `_donate-components.scss`. Mainly to avoid updating the `donation-circles.html` page since I would consider it outside the scope of the issue.
  - [x] I would also rename `.override-p-padding` so the name aligns more with the other classes in the SCSS file. You can add a comment explaining what the class is for. 
  - [x] There isn't an updated Figma wire frame to go off of, but I prefer to have the `padding: 0.2em 0`. This is to center the text in the circle more and more closely match the current design on the website.
- [x] Could you move the images from the pdf files to this comment? Desktop and mobile views are sufficient. This is mainly to follow Hack for LA's documentation practices.
- [x] Could you update the pull request name to fix the spelling mistake and remove the issue#?

Adastros commented on _includes/join-donate-card.html:
```diff
@@ -10,8 +10,10 @@ <h3 class="title6 page-card--large-icon-header-secondary">Running Hack for LA co
-      </p>
-      <img class="join-us-donation-img" src="/assets/images/join-us/donation-graphic.svg" alt="Monthly: Hosting, Mailing List services, zoom; Annual: Domain Names; Occasional: Stickers, T-shirts" />
+      </p>      
+      <div class="join-us-donation-circles">
```
One more thing, could you explain the addition of this div? It doesn't seem like the issue called out for it specifically and the `join-us-donation-circles` class doesn't seem to be used anywhere else. The `donation-circles.html` file seems to be working fine without the parent div.

jenny-alexander replied:
I removed the div.

roslynwythe commented:
Availability: 5/25 10-2
ETA EOD 5/26

roslynwythe reviewed (changes requested):
Thank you @jenny-alexander for outstanding work on this PR and to @one2code and @Adastros for great reviews!  When writing this issue I did not anticipate the need to update donation-circles.html but I have no objection to the solution implemented in this PR, and I would request that you remove the duplicate padding in _donate-components.scss to resolve the linter error.  I added an Action Item in the issue to make it clear that your updates to donatation-circles.html are within the scope of the issue and I apologize for not making that clear to begin with.  

I would approve either the solution implemented in the PR or the clever sass for-loop solution proposed by one2code.   

I agree with other changes requested by Adastros.  

@jenny-alexander thank you for setting up the branches correctly, describing your work really well, making clean code changes and doing thorough testing on both the Donate and Join Us pages.

jenny-alexander commented:
@one2code , thank you for your review.

I'm not familiar with sass for loops. Can you tell me where (in the _donate-components.scss file) you added the for loop?

I added it to the bottom of the _donate-components.scss file and received this error.

hfla_site  |   Conversion error: Jekyll::Converters::Scss encountered an error while converting 'assets/css/main.scss':
hfla_site  |                     Undefined variable: "$inside-circle-items". on line 80
hfla_site  |              Error: Undefined variable: "$inside-circle-items". on line 80
hfla_site  |              Error: Run jekyll build --trace for more information.

jenny-alexander committed:
Merge branch 'upstream-gh-pages' into integrate-donation-cicle-with-join-us-include-4544

jenny-alexander committed:
Remove padding 0 causing lint error & add comment to css id selector

jenny-alexander commented:
I made a few changes as requested:

1. Fix spelling error in PR name and removed fix number from PR name.
2. Made selector id name more descriptive and added a comment above it.
3. Removed duplicate padding in _donate-components.scss that was causing linter error.
4. Added 'before' and 'after' images directly to PR initial comment.

I did not implement the for sass loop as I couldn't get it to work on my end and I felt adding the id selector was quite straightforward.
I also kept the padding to 0.4em 0 because making it 0.2em 0 (as requested by @Adastros ) altered it a bit from the original.

one2code commented:
No worries @jenny-alexander. I'll review your latest changes and submit my review. I think you did an excellent job of troubleshooting and resolving the unexpected problem you encountered.

As for the for loop, I forgot to include a closing brace in my example. Here is the correct example.
`
@for $i from 1 through length($inside-circle-items) {
    .inside-circle-text:nth-child(#{$i}) {
      content: nth($inside-circle-items, $i);
    }
    }`

one2code reviewed (changes requested):
I've reviewed and tested your latest changes. I noticed that the padding on #circle-override-p-padding wasn't updated to the requested 0.2em that @Adastros and @roslynwythe mentioned. Since that is the new agreed upon unit, I will request that the padding be changed on that id. Great job on addressing the other requested changes by @Adastros, such as renaming the id to a more specific name in-line with the codebase.

jenny-alexander committed:
Make padding 0.2em instead of 0.4em

jenny-alexander commented:
@one2code I adjusted the padding to 0.2em as requested.

one2code reviewed (dismissed):
@jenny-alexander Thank you for making the requested changes to the #circle-override-p-padding. Good job on resolving the issues you came across in this PR.

roslynwythe commented:
Thank you @jenny-alexander for making all of the requested changes.  I checked the PR branch in Chrome and it looks really great.  In the "Visuals after changes are applied" section, the desktop image isn't displaying.  Could you fix that, and then I'll approve/merge this?

jenny-alexander commented:
@roslynwythe - I fixed the initial comment so that the desktop "after" visual appears!

Adastros reviewed (changes requested):
Hi @jenny-alexander, thank you for implementing the changes per my last comment! The new class name aligns with the other class names for the `_donate-components.scss` files. I'm okay with not adding the SASS loop as proposed by @one2code based on clarification from @roslynwythe. 

Could you:

- [x] Update the pull request comment to update the "What changes did you make and why did you make them?" section based on the latest change made? 
- [x] Update the "Join Us - Desktop Breakpoint" image in the pull request comment? The screenshot shows the donation circles as left-aligned instead of centered per the updates made in this pull request. 
- [x] The Join us testing pdf will need to be updated for the same issue above.
- [x] I left a pending review comment on the `<div class="join-us-donation-circles">` element. Could you address this? Please remove this div if it's not needed. The liquid code is sufficient for the changes to work since it's self contained.

jenny-alexander committed:
Merge branch 'upstream-gh-pages' into integrate-donation-cicle-with-join-us-include-4544

jenny-alexander committed:
Merge branch 'upstream-gh-pages' into integrate-donation-cicle-with-join-us-include-4544

jenny-alexander committed:
remove <div>

jenny-alexander commented:
@Adastros changes you requested were done. Please review.

Adastros reviewed (dismissed):
Hi @jenny-alexander, the information in the pull request comment has references to `<div class="join-us-donation-circles">` which was removed and the old id name of `#override-p-padding`. Can you update these to reflect the latest changes? 

Also, I didn't notice this earlier but per the browser developer tools it seems that removing the `padding: 0` from the `.circle-full-container` class causes the annual circle to be cropped on the side for certain mobile devices. This seen using the Chrome and Firefox browser tools.

<details>

![image](https://github.com/hackforla/website/assets/48114590/23cb9b9a-173e-4646-91d9-d63992cc7257)

</details>

Can you add the following to the `.circle-full-container` class. This will fix the issue for mobile views. Please update the pull request comment to mention this as well. 

```
@media #{$bp-below-tablet} {
        padding: 0;
}
```

roslynwythe reviewed (approved):
Thank you @jenny-alexander for great work and for being responsive and accommodating to requests for changes on this PR.
